### PR TITLE
[ENH] Msm distance cleanup

### PR DESCRIPTION
**Pull request by chrisholder**
This PR cleans up msm redundent functions and improves docstrong

#### What does this implement/fix? Explain your changes.

MSM was missing docstrings for variables and additionally there was redundent functions in the file which have been removed.
Additionally code has been simplified using built in python functions.

##### For all contributions
- [x] I've added myself to the [list of contributors](https://github.com/alan-turing-institute/sktime/blob/main/.all-contributorsrc).
- [x] Optionally, I've updated sktime's [CODEOWNERS](https://github.com/alan-turing-institute/sktime/blob/main/CODEOWNERS) to receive notifications about future changes to these files.
- [x] I've added unit tests and made sure they pass locally.
- [x] The PR title starts with either [ENH], [MNT], [DOC], or [BUG] indicating whether the PR topic is related to enhancement, maintenance, documentation, or bug.

##### For new estimators
- [x] I've added the estimator to the online documentation.
- [x] I've updated the existing example notebooks or provided a new one to showcase how my estimator works.

### CODE REVIEW

File: sktime/classification/distance_based/tests/test_time_series_neighbors.py [hunk 1]
```diff
@@ -26,6 +26,17 @@
...
+    "wdtw": 21,
+    "msm": 10,
+    "erp": 19,
+    "edr": 20,
+    "lcss": 12,
+}
+
 
 def test_knn_on_unit_test():
     """Test function for elastic knn, to be reinstated soon."""
```
Comment (TonyBagnall) [inline sktime/classification/distance_based/tests/test_time_series_neighbors.py:42]: we can remove the "to be reinstated soon"?

File: sktime/distances/tests/_expected_results.py [hunk 2]
```diff
@@ -13,7 +13,7 @@
     "ddtw": [0.0, 2.0884818837222006, 34.837800040564005],
     "wdtw": [12.343758137512241, 0.985380547171357, 21.265839226825413],
     "wddtw": [0.0, 1.0442409418611003, 17.418900020282003],
-    "msm": [5.0, 3.0922226886554434, None],
+    "msm": [5.0, 6.828557434224288, None],
```
Comment (TonyBagnall) [inline sktime/distances/tests/_expected_results.py:16]: why have the expected results changed? Is it the different default c value? Can we also replace the None?

**Review by TonyBagnall — CHANGES_REQUESTED** [review]
so minor points.

**Review by TonyBagnall — APPROVED** [review]

**Review by TonyBagnall — DISMISSED** [review]

**Review by fkiraly — APPROVED** [review]
Looks like sensible changes.

**Pull request merged**